tfhe/src/integer/server_key/radix_parallel/tests_signed/test_rotate.rs

```diff
@@ -0,0 +1,485 @@
+use crate::integer::keycache::KEY_CACHE;
+use crate::integer::server_key::radix_parallel::tests_cases_unsigned::FunctionExecutor;
+use crate::integer::server_key::radix_parallel::tests_unsigned::CpuFunctionExecutor;
+use crate::integer::tests::create_parametrized_test;
+use crate::integer::{
+    IntegerKeyKind, RadixCiphertext, RadixClientKey, ServerKey, SignedRadixCiphertext,
+};
+#[cfg(tarpaulin)]
+use crate::shortint::parameters::coverage_parameters::*;
+use crate::shortint::parameters::*;
+use rand::prelude::ThreadRng;
+use rand::Rng;
+use std::sync::Arc;
+
+/// Number of loop iteration within randomized tests
+#[cfg(not(tarpaulin))]
+pub(crate) const NB_TESTS: usize = 30;
+/// Smaller number of loop iteration within randomized test,
+/// meant for test where the function tested is more expensive
+#[cfg(not(tarpaulin))]
+pub(crate) const NB_TESTS_SMALLER: usize = 10;
+
+// Use lower numbers for coverage to ensure fast tests to counter balance slowdown due to code
+// instrumentation
+#[cfg(tarpaulin)]
+pub(crate) const NB_TESTS: usize = 1;
+#[cfg(tarpaulin)]
+pub(crate) const NB_TESTS_SMALLER: usize = 1;
+/// Unchecked test cases needs a minimum number of tests of 4 in order to provide guarantees.
+#[cfg(tarpaulin)]
+pub(crate) const NB_TESTS_UNCHECKED: usize = 4;
+
+#[cfg(not(tarpaulin))]
+pub(crate) const NB_CTXT: usize = 4;
+#[cfg(tarpaulin)]
+pub(crate) const NB_CTXT: usize = 2;
+
+create_parametrized_test!(integer_signed_unchecked_left_rotate);
+create_parametrized_test!(integer_signed_unchecked_right_rotate);
+create_parametrized_test!(integer_signed_left_rotate);
+create_parametrized_test!(integer_signed_right_rotate);
+
+pub(crate) fn random_non_zero_value(rng: &mut ThreadRng, modulus: i64) -> i64 {
```

Comment:
This function should go to radix_parallel/tests_sigfned/mod.rs to be reused across tests.

Reply:
I see!
I've moved them back.